com.teamscale.polarion.plugin/README.md

```diff
@@ -0,0 +1,148 @@
+
+**Items moved to the recycle bin (soft deletion):** In Polarion, users can soft delete items by moving them to the [recycle bin](https://docs.plm.automation.siemens.com/content/polarion/19.3/help/en_US/user_and_administration_help/user_guide/work_with_documents/work_items_in_documents/work_item_recycle_bin.html). These items do not show up in documents (only if you open the recycle bin UI) but they're still valid items when we query the work item table in Polarion database (and consequently they're still related to the document in the database). As of now, the plugin only detects work item deletion if items are in the recycle bin. For [hard deletions](https://docs.plm.automation.siemens.com/content/polarion/20/help/en_US/user_and_administration_help/user_guide/work_items/work_item_actions/delete_work_items.html), the plugin is not currently able to detectem them. See next.
+
+**Items deleted permanently:** We have not found a way to query [permanently deleted items](https://docs.plm.automation.siemens.com/content/polarion/20/help/en_US/user_and_administration_help/user_guide/work_items/work_item_actions/delete_work_items.html) utilizing the Polarion Java API. Therefore, the current solution is to leave that job to the client side. For that reason, every response the plugin sends back to the client contains a list of all work item ids that are valid (not deleted) at the moment. On the client side, applications can build some logic around that. For instance:
+ - Client builds a set (A) of item ids known to the client before request
+ - Client sends request and takes the set (B) of all item ids returned in the response field 'allItemsIds' (all items not deleted at the moment)
+ - Client performs diff (A) - (B). If the result is empty, no items were deleted.
+   - If the result is not empty, the remaining items are the deleted items since lastUpdate revision. 
+   - Note that (B) can have more items than (A) if new items were created since lastUpdate revision.
+
+**Thread Safety:**
+The solution is thread safe. This Polarion plugin is implemented as a Java Servlet. Polarion utilizes the Apache Tomcat web container to run servlets. The Tomcat web container follows a multi-thread model, meaning that it creates a new thread per request directed to a any given servlet. Besides that, in the plugin, we do not utilize any shared data structures across different servlets from Polarion. Each request to the plugin servlet will run in a separate thread, which will query the database independently and work in its own data in memory.
+
+## JSON Serialization
+We currently use the opensource library [Gson](https://github.com/google/gson) which already comes available in the Polarion installation.
```

Comment:
I don't think there's a BOM. I looked around and didn't find anything like that. 

I didn't determine any specific GSON version. It just becomes automatically available in the project once you declare it in the code, simply because it's already 'pre-loaded' together with all other Polarion dependencies. Inside this path <polarion-installation>/polarion/plugins you can find all the dependencies that become automatically pre-loaded when you're developing a plugin. GSON is one of them (and the version 2.8.9.patched is the one for this specific Polarion dist). So, basically, the plugin developer doesn't get to choose the version, unless we force a different class loader.